lib/timers.js

```diff
@@ -409,6 +393,26 @@ function processImmediate() {
   }
 }
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw) {
```

Comment:
I also suggest we use the same style like https://github.com/nodejs/node/pull/4007/files#diff-0a5d4868b2b9b17cf9e2c11f1bd1311eR185 i.e. `if (!threw) return;`